.github/workflows/build.yml

```diff
@@ -0,0 +1,75 @@
+name: Build and Deploy
+on: [push]
+# TODO: limit this flow to when we push to main
```

Comment:
I think we need to change this setting to only deploy the site when we push to main, instead of every push